Update checkpointing documentation to mark resume_from_checkpoint as deprecated (#20361)

WangYue0000 opened the pull request:
This PR updates the checkpointing documentation to mark `resume_from_checkpoint` as deprecated 
and provides examples of the recommended `ckpt_path` method in the `Trainer.fit()` function.

Changes made:
- Added a warning box in the "Resume training state" section to discourage the use of `resume_from_checkpoint`.
- Updated examples to reflect the correct usage of `ckpt_path`.
- Ensured consistency across the documentation by replacing all occurrences of the deprecated parameter.

Fixes #20361.

<!-- readthedocs-preview pytorch-lightning start -->
----
📚 Documentation preview 📚: https://pytorch-lightning--20477.org.readthedocs.build/en/20477/

<!-- readthedocs-preview pytorch-lightning end -->

WangYue0000 commented:
@Borda @lantiga Hi！Could you please take a look at this PR when you get a chance? It updates the checkpointing documentation to address #20361. Thanks, hope my solution helps!

lantiga commented on docs/source-pytorch/common/checkpointing_basic.rst:
```diff
@@ -20,6 +20,13 @@ PyTorch Lightning checkpoints are fully usable in plain PyTorch.
+
+   **Important Update: Deprecated Method**
+
+   Starting from PyTorch Lightning v1.0.0, the `resume_from_checkpoint` parameter has been deprecated. To resume training from a checkpoint, use the `ckpt_path` parameter in the `fit()` method.
```
```suggestion
   Starting from PyTorch Lightning v1.0.0, the `resume_from_checkpoint` argument has been deprecated. To resume training from a checkpoint, use the `ckpt_path` argument in the `fit()` method.
```

lantiga commented on docs/source-pytorch/common/checkpointing_basic.rst:
```diff
@@ -197,16 +204,31 @@ You can disable checkpointing by passing:
 
+.. warning::
+
+   The parameter `resume_from_checkpoint` has been deprecated in recent versions of PyTorch Lightning.
```
```suggestion
   The argument `resume_from_checkpoint` has been deprecated in versions of PyTorch Lightning >= 1.0.0.
```

lantiga replied:
Also, since this will now come after the correct way, I'd word this as:

In versions of PyTorch Lightning prior to 1.0.0, resuming training state was achieved via the `resume_from_checkpoint` argument of `Trainer`. The argument has been deprecated since, in favor of the `ckpt_path` argument of `fit`.

lantiga commented on docs/source-pytorch/common/checkpointing_basic.rst:
```diff
@@ -197,16 +204,31 @@ You can disable checkpointing by passing:
 
 If you don't just want to load weights, but instead restore the full training, do the following:
 
+.. warning::
```
Can we move the warning after the correct usage? Let's first describe the right way to go about it, then show the warning.

lantiga reviewed (commented):
Thank you for your contribution! I added a couple of comments

WangYue0000 commented:
@lantiga Thank you for the detailed feedback! I've addressed all your suggestions, including:
- Updated terminology from "parameter" to "argument."
- Added version details for the deprecation.
- Moved the warning box to appear after the correct usage.
- Improved the phrasing as per your recommendation.

Please let me know if there are any additional changes needed!

lantiga reviewed (approved):
Looks great, thank you @WangYue0000 !
